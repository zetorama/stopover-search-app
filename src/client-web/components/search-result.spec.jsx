import React from 'react'
import { mount } from 'enzyme'
import Comp from './search-result'

import offersData from '../../__mock__/hotel-offers.json'

describe('search-result comp', () => {
  test('renders empty result by default', () => {
    const res = mount(<Comp />)
    expect(res).toMatchSnapshot()
  })

  test('renders empty result when offers are empty', () => {
    const res = mount(<Comp offers={[]} />)
    expect(res).toMatchSnapshot()
  })

  test('renders hotel offers', () => {
    const res = mount(<Comp offers={offersData} />)
    expect(res).toMatchSnapshot()
  })
})
