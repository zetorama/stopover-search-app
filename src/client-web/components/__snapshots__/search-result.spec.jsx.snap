// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-result comp renders empty result by default 1`] = `
<SearchResult>
  <EmptyResult>
    <Note
      ctx="warning"
      title="Cannot find any offers"
    >
      <div
        className="alert alert-warning"
        role="note"
      >
        <h4
          className="alert-heading"
        >
          Cannot find any offers
        </h4>
        <p>
          Please, try to adjust your search criteria.
        </p>
      </div>
    </Note>
  </EmptyResult>
</SearchResult>
`;

exports[`search-result comp renders empty result when offers are empty 1`] = `
<SearchResult
  offers={Array []}
>
  <EmptyResult>
    <Note
      ctx="warning"
      title="Cannot find any offers"
    >
      <div
        className="alert alert-warning"
        role="note"
      >
        <h4
          className="alert-heading"
        >
          Cannot find any offers
        </h4>
        <p>
          Please, try to adjust your search criteria.
        </p>
      </div>
    </Note>
  </EmptyResult>
</SearchResult>
`;

exports[`search-result comp renders hotel offers 1`] = `
<SearchResult
  offers={
    Array [
      Object {
        "address": Object {
          "cityName": "PARIS",
          "countryCode": "FR",
          "lines": Array [
            "4 ALLEE DU VERGER",
          ],
          "postalCode": "95700",
        },
        "contact": Object {
          "fax": "33-1-30182201",
          "phone": "33-1-30182200",
        },
        "id": "XKCDG0E0",
        "isAvailable": true,
        "name": "HOLIDAY INN PARIS CDG AIRPORT",
        "offers": Array [
          Object {
            "currency": "USD",
            "id": "8060D9C4598956A01752F8B0C4CC4F2A80ECBA000D1A6DA49AA86918D36D415A",
            "roomType": Object {
              "bedType": "QUEEN",
              "beds": 2,
              "category": "STANDARD_ROOM",
            },
            "totalPrice": "79.99",
          },
        ],
      },
      Object {
        "address": Object {
          "cityName": "ROISSY CDG CEDEX",
          "countryCode": "FR",
          "lines": Array [
            "ROISSYPOLE OUEST",
            "ROUTE DE LA COMMUNE BP 20248",
          ],
          "postalCode": "95713",
        },
        "contact": Object {
          "email": "h0577@accor.com",
          "fax": "33/1/49192900",
          "phone": "33/1/49192929",
        },
        "id": "RTCDGAIR",
        "isAvailable": true,
        "name": "MERCURE PARIS CDG AIRPORT",
        "offers": Array [
          Object {
            "currency": "EUR",
            "id": "D5DE80B70D60BE00543A6AB76640663B1828EFE4660CE2F88D7416EFD63C3E14",
            "roomType": Object {
              "bedType": "DOUBLE",
              "beds": 1,
              "category": "STANDARD_ROOM",
            },
          },
        ],
      },
    ]
  }
>
  <HotelOffers
    offers={
      Array [
        Object {
          "address": Object {
            "cityName": "PARIS",
            "countryCode": "FR",
            "lines": Array [
              "4 ALLEE DU VERGER",
            ],
            "postalCode": "95700",
          },
          "contact": Object {
            "fax": "33-1-30182201",
            "phone": "33-1-30182200",
          },
          "id": "XKCDG0E0",
          "isAvailable": true,
          "name": "HOLIDAY INN PARIS CDG AIRPORT",
          "offers": Array [
            Object {
              "currency": "USD",
              "id": "8060D9C4598956A01752F8B0C4CC4F2A80ECBA000D1A6DA49AA86918D36D415A",
              "roomType": Object {
                "bedType": "QUEEN",
                "beds": 2,
                "category": "STANDARD_ROOM",
              },
              "totalPrice": "79.99",
            },
          ],
        },
        Object {
          "address": Object {
            "cityName": "ROISSY CDG CEDEX",
            "countryCode": "FR",
            "lines": Array [
              "ROISSYPOLE OUEST",
              "ROUTE DE LA COMMUNE BP 20248",
            ],
            "postalCode": "95713",
          },
          "contact": Object {
            "email": "h0577@accor.com",
            "fax": "33/1/49192900",
            "phone": "33/1/49192929",
          },
          "id": "RTCDGAIR",
          "isAvailable": true,
          "name": "MERCURE PARIS CDG AIRPORT",
          "offers": Array [
            Object {
              "currency": "EUR",
              "id": "D5DE80B70D60BE00543A6AB76640663B1828EFE4660CE2F88D7416EFD63C3E14",
              "roomType": Object {
                "bedType": "DOUBLE",
                "beds": 1,
                "category": "STANDARD_ROOM",
              },
            },
          ],
        },
      ]
    }
  >
    <div
      className="card-group mb-3"
    >
      <Hotel
        address={
          Object {
            "cityName": "PARIS",
            "countryCode": "FR",
            "lines": Array [
              "4 ALLEE DU VERGER",
            ],
            "postalCode": "95700",
          }
        }
        contact={
          Object {
            "fax": "33-1-30182201",
            "phone": "33-1-30182200",
          }
        }
        id="XKCDG0E0"
        isAvailable={true}
        key="XKCDG0E0"
        name="HOLIDAY INN PARIS CDG AIRPORT"
        offers={
          Array [
            Object {
              "currency": "USD",
              "id": "8060D9C4598956A01752F8B0C4CC4F2A80ECBA000D1A6DA49AA86918D36D415A",
              "roomType": Object {
                "bedType": "QUEEN",
                "beds": 2,
                "category": "STANDARD_ROOM",
              },
              "totalPrice": "79.99",
            },
          ]
        }
      >
        <div
          className="card"
        >
          <div
            className="card-header"
          >
            <span
              title="Available"
            >
              ‚úÖ
            </span>
            <strong>
              HOLIDAY INN PARIS CDG AIRPORT
            </strong>
          </div>
          <ul
            className="list-group list-group-flush"
          >
            <li
              className="list-group-item"
              key="8060D9C4598956A01752F8B0C4CC4F2A80ECBA000D1A6DA49AA86918D36D415A"
            >
              <Room
                currency="USD"
                id="8060D9C4598956A01752F8B0C4CC4F2A80ECBA000D1A6DA49AA86918D36D415A"
                roomType={
                  Object {
                    "bedType": "QUEEN",
                    "beds": 2,
                    "category": "STANDARD_ROOM",
                  }
                }
                totalPrice="79.99"
              >
                <span
                  className="badge badge-success"
                >
                  USD
                   
                  79.99
                </span>
                 
                <RoomType
                  bedType="QUEEN"
                  beds={2}
                  category="STANDARD_ROOM"
                >
                  STANDARD_ROOM
                   
                  <br
                    className="d-lg-none"
                  />
                  2 QUEEN beds
                </RoomType>
              </Room>
            </li>
            <li
              className="list-group-item"
              title="Contact information"
            >
              <Contact
                fax="33-1-30182201"
                phone="33-1-30182200"
              >
                ‚òéÔ∏è
                +33-1-30182200
              </Contact>
            </li>
            <li
              className="list-group-item"
              title="Address"
            >
              <Address
                cityName="PARIS"
                countryCode="FR"
                lines={
                  Array [
                    "4 ALLEE DU VERGER",
                  ]
                }
                postalCode="95700"
              >
                üè®
                4 ALLEE DU VERGER
                <br />
                PARIS
                 (
                FR
                ) 
                95700
              </Address>
            </li>
          </ul>
        </div>
      </Hotel>
      <Hotel
        address={
          Object {
            "cityName": "ROISSY CDG CEDEX",
            "countryCode": "FR",
            "lines": Array [
              "ROISSYPOLE OUEST",
              "ROUTE DE LA COMMUNE BP 20248",
            ],
            "postalCode": "95713",
          }
        }
        contact={
          Object {
            "email": "h0577@accor.com",
            "fax": "33/1/49192900",
            "phone": "33/1/49192929",
          }
        }
        id="RTCDGAIR"
        isAvailable={true}
        key="RTCDGAIR"
        name="MERCURE PARIS CDG AIRPORT"
        offers={
          Array [
            Object {
              "currency": "EUR",
              "id": "D5DE80B70D60BE00543A6AB76640663B1828EFE4660CE2F88D7416EFD63C3E14",
              "roomType": Object {
                "bedType": "DOUBLE",
                "beds": 1,
                "category": "STANDARD_ROOM",
              },
            },
          ]
        }
      >
        <div
          className="card"
        >
          <div
            className="card-header"
          >
            <span
              title="Available"
            >
              ‚úÖ
            </span>
            <strong>
              MERCURE PARIS CDG AIRPORT
            </strong>
          </div>
          <ul
            className="list-group list-group-flush"
          >
            <li
              className="list-group-item"
              key="D5DE80B70D60BE00543A6AB76640663B1828EFE4660CE2F88D7416EFD63C3E14"
            >
              <Room
                currency="EUR"
                id="D5DE80B70D60BE00543A6AB76640663B1828EFE4660CE2F88D7416EFD63C3E14"
                roomType={
                  Object {
                    "bedType": "DOUBLE",
                    "beds": 1,
                    "category": "STANDARD_ROOM",
                  }
                }
              >
                <RoomType
                  bedType="DOUBLE"
                  beds={1}
                  category="STANDARD_ROOM"
                >
                  STANDARD_ROOM
                   
                  <br
                    className="d-lg-none"
                  />
                  1 DOUBLE bed
                </RoomType>
              </Room>
            </li>
            <li
              className="list-group-item"
              title="Contact information"
            >
              <Contact
                email="h0577@accor.com"
                fax="33/1/49192900"
                phone="33/1/49192929"
              >
                ‚òéÔ∏è
                +33-1-49192929
              </Contact>
            </li>
            <li
              className="list-group-item"
              title="Address"
            >
              <Address
                cityName="ROISSY CDG CEDEX"
                countryCode="FR"
                lines={
                  Array [
                    "ROISSYPOLE OUEST",
                    "ROUTE DE LA COMMUNE BP 20248",
                  ]
                }
                postalCode="95713"
              >
                üè®
                ROISSYPOLE OUEST
                <br />
                ROUTE DE LA COMMUNE BP 20248
                <br />
                ROISSY CDG CEDEX
                 (
                FR
                ) 
                95713
              </Address>
            </li>
          </ul>
        </div>
      </Hotel>
    </div>
  </HotelOffers>
</SearchResult>
`;
